src/gcp/cloudscheduler.ts

```diff
@@ -12,38 +14,116 @@ export interface Schedule {
     uri: string;
     httpMethod: string;
   };
+  retryConfig?: {
+    retryCount?: number;
+    maxRetryDuration?: string;
+    minBackoffDuration?: string;
+    maxBackoffDuration?: string;
+    maxDoublings?: number;
+  };
 }
 
-export function createJob(schedule: Schedule): Promise<void> {
+/**
+ * Creates a cloudScheduler job.
+ * If another job with that name already exists, this will return a 409.
+ * @param job The job to create.
+ */
+export function createJob(job: Job): Promise<any> {
   // the replace below removes the portion of the schedule name after the last /
   // ie: projects/my-proj/locations/us-central1/jobs/firebase-schedule-func-us-east1 would become
   // projects/my-proj/locations/us-central1/jobs
-  const strippedName = schedule.name.replace(/\/[^\/]+$/, "");
+  const strippedName = job.name.substring(0, job.name.lastIndexOf("/"));
   return api.request("POST", `/${VERSION}/${strippedName}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
-    data: Object.assign({ timeZone: "America/Los_Angeles" }, schedule),
+    data: Object.assign({ timeZone: DEFAULT_TIME_ZONE }, job),
   });
 }
 
-export function deleteJob(name: string): Promise<void> {
+/**
+ * Deletes a cloudScheduler job with the given name.
+ * Returns a 404 if no job with that name exists.
+ * @param name The name of the job to delete.
+ */
+export function deleteJob(name: string): Promise<any> {
   return api.request("DELETE", `/${VERSION}/${name}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
-export async function createOrReplaceJob(schedule: Schedule): Promise<void> {
-  const jobName = `${schedule.name.split("/")[5]}`;
+/**
+ * Gets a cloudScheduler job with the given name.
+ * If no job with that name exists, this will return a 404.
+ * @param name The name of the job to get.
+ */
+export function getJob(name: string): Promise<any> {
+  return api.request("GET", `/${VERSION}/${name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+  });
+}
+
+/**
+ * Updates a cloudScheduler job.
+ * Returns a 404 if no job with that name exists.
+ * @param job A job to update. Note that name cannot be updated.
+ */
+export function updateJob(job: Job): Promise<any> {
+  return api.request("PATCH", `/${VERSION}/${job.name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+    data: Object.assign({ timeZone: DEFAULT_TIME_ZONE }, job),
+  });
+}
+
+/**
+ * Checks for a existing job with the given name.
+ * If none is found, it creates a new job.
+ * If one is found, and it is identical to the job parameter, it does nothing.
+ * Otherwise, if one is found and it is different from the job param, it updates the job.
+ * @param job A job to check for and create, replace, or leave as appropriate.
+ * @throws { FirebaseError } if an error response other than 404 is received on the GET call.
+ */
+export async function createOrReplaceJob(job: Job): Promise<any> {
+  const jobName = `${job.name.split("/").slice(-1)}`;
```

Comment:
Why `slice(-1)` over `pop()`? I don't have to think about what `pop` does... `slice(-1)` takes me a second